### PR TITLE
terraform deployment with seedfarmer

### PR DESCRIPTION
**Pull request by srinivasreddych**
Deploying terraform IAC using seedfarmer

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: modules/examples/example-tf/main.tf [hunk 1]
```diff
@@ -0,0 +1,29 @@
+# Provider Configuration
+provider "aws" {
+  region = var.aws_region
+  access_key = ""
+  secret_key = ""
```
Comment (a13zen) [inline modules/examples/example-tf/main.tf:5]: We can drop these right? You don't need them afaik?

File: modules/examples/example-tf/deployspec.yaml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+        - echo dynamodb_table=\"$ADDF_PARAMETER_TF_DDB_TABLE\" >> vars/backend.tfvars
+        - cat vars/backend.tfvars
+        # App specific vars configuration
+        - echo aws_region=\"$AWS_DEFAULT_REGION\" >> vars/variables.tfvars
+        - cat vars/variables.tfvars
+    build:
+      commands:
+       - terraform init -input=false -backend=true -backend-config=vars/backend.tfvars -reconfigure
+       - terraform plan -input=false -out tf.plan -var-file=vars/variables.tfvars -no-color
+       - terraform apply -auto-approve -var-file=vars/variables.tfvars
```
Comment (a13zen) [inline modules/examples/example-tf/deployspec.yaml:24]: Can we export outputs back to seedfarmer?

File: modules/examples/example-tf/modulestack.yaml [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            Resource: 
+              - !Sub "arn:aws:s3:::addf-tf*/${DeploymentName}/${ModuleName}*"
+          - Effect: Allow
+            Action:
+              - "dynamodb:DescribeTable"
+              - "dynamodb:GetItem"
+              - "dynamodb:PutItem"
+              - "dynamodb:DeleteItem"
+            Resource: 
+              - !Sub "arn:aws:dynamodb:${AWS::Region}:${AWS::AccountId}:table/addf-tf*"
```
Comment (a13zen) [inline modules/examples/example-tf/modulestack.yaml:41]: What creates the Dynamo and State bucket here? Manual for now?

Comment (srinivasreddych) [reply modules/examples/example-tf/modulestack.yaml:41]: Manual for now, but i am working on a cdk stack which creates those resources (if someone wants to use them)

File: modules/examples/example-tf/main.tf (hunk 1 above)
Comment (srinivasreddych) [reply modules/examples/example-tf/main.tf:5]: which ones you meant?

File: modules/examples/example-tf/deployspec.yaml (hunk 2 above)
Comment (srinivasreddych) [reply modules/examples/example-tf/deployspec.yaml:24]: Done. I used `terraform output -json` and wrote to a `json` file. The contents of the json file will be quite noisy with some unnecessary elements in each resource's outputs.

Ex: 
```output
ADDF_MODULE_METADATA: {'s3_bucket_id': {'sensitive': False, 'type': 'string', 'value': 'example-XX-us-east-1-XX'}}
```

Comment (srinivasreddych) [reply modules/examples/example-tf/deployspec.yaml:24]: Now, we need to figure out how would a downstream module depending on this module would consume the values, eliminating the unnecessary attributes

File: modules/examples/example-tf/modulestack.yaml [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+    Type: "AWS::IAM::Policy"
+    Properties:
+      PolicyDocument:
+        Statement:
+          # Terraform backend specific permissions
+          - Action:
+              - "s3:List*"
+            Effect: Allow
+            Resource: 
+              - "arn:aws:s3:::addf-tf*"
```
[the remark this answers is not in the document]
Comment (srinivasreddych) [reply modules/examples/example-tf/modulestack.yaml]: good catch. addressed it

File: modules/examples/example-tf/deployspec.yaml (hunk 2 above)
Comment (a13zen) [reply modules/examples/example-tf/deployspec.yaml:24]: This would be a really good case for an seedfarmer util right?

**Review by a13zen — APPROVED** [review]
LGTM

**Review by dgraeber — APPROVED** [review]

**Pull request merged**